server/autolinkplugin/plugin.go

```diff
@@ -92,6 +124,27 @@ func (p *Plugin) ProcessPost(c *plugin.Context, post *model.Post) (*model.Post,
 	conf := p.getConfig()
 	postText := post.Message
 	offset := 0
+
+	hasOneOrMoreScopes := false
+	for _, link := range conf.Links {
+		if len(link.Scope) > 0 {
+			hasOneOrMoreScopes = true
+			break
+		}
+	}
+
+	channelName := ""
+	teamName := ""
+	if hasOneOrMoreScopes {
+		cn, tn, rsErr := p.resolveScope(post.ChannelId)
+		channelName = cn
+		teamName = tn
+
+		if rsErr != nil {
+			p.API.LogError("Failed to resolve scope", "error", rsErr.Error())
+		}
+	}
+
```

Comment:
May be we could put this logic inside `p.resolveScope`? I think it will be clearer.

Reply:
I'm having a problem agreeing with you. This attempt might appear that it will make the logic clearer but in fact it's more misleading:

1. It's not the responsibility of the resolve function to check when to resolve or not
2. It's the responsibility of the `ProcessPost` function to decide when to resolve or not, thus the logic is in it's right place
3. If we hide this logic inside the resolve it will appear misleading as the user will expect that the function resolves right away as per it's name

what do yo think?

Reply:
Makes sense @mo2menelzeiny, I would have this logic in a different method anyways, but it's not a big deal and it's your call.